src/stories/Users/CanDeleteAccount/story.js

```diff
@@ -0,0 +1,55 @@
+const attributesRepo = requireRepo("attributes");
+const tokensRepo = requireRepo("tokens");
+
+const prepare = ({ reqQuery, reqBody, reqParams, req }) => {
+  return {
+    jti: req.jti,
+  };
+};
+
+const authorize = async ({ prepareResult }) => {
+  let unauthorizedObject = {
+    statusCode: 401,
+    message: "Unauthorized",
+  };
+
+  try {
+    if (prepareResult.jti !== undefined) {
+      let token = await tokensRepo.first({ uuid: prepareResult.jti });
+      if (token["issuer"] === "user") {
+        return token;
+      } else {
+        throw unauthorizedObject;
+      }
+    } else {
+      throw unauthorizedObject;
+    }
+  } catch (error) {
+    throw error;
+  }
+};
+
+const handle = async ({ prepareResult, authorizeResult }) => {
+  try {
+    return await attributesRepo.deleteAccount({
```

Comment:
@ShubhamJoshi-1994 I think we should also "Logout" the user once they are deleted, that's what got missed out, one way to do it would be:

```
tokensRepo.deleteTokenByConstraints({
      uuid: prepareResult.jti,
    });
```

But this might not be the correct way, because user might have multiple tokens, we should delete all the tokens for that user from tokens table.

If we call something like the following:

```

```
tokensRepo.deleteTokenByConstraints({
      sub: "user's uuid",
      issuer: "user"
    });
```

The above should delete all tokens issued for that user. 

The point you highlighted, /api/user working would also be fixed with it.

Reply:
ok, @rjvim I'll implement this and test the scenario.

Reply:
@rjvim I have implemented the above code and it's deleting the tokens also now.

After that Get user API gives `unauthenticated` in response.